### DISCUSSION TITLE
intercept GitHub 404 page

### DISCUSSION THREAD

**Discussion by DG12** [community/community#33104] [Other Feature Feedback, Questions, & Ideas]
I tried to follow instructions at

https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-custom-404-page-for-your-github-pages-site
1) I created 404.md containing
```
---
permalink: /404.html
---
```
2) I created 404.html containing 

```
<!DOCTYPE html>
<head>
<title>404 Page</title>
</head>
<body>
 simple 404 error page for GitHub Pages 
</body>
</html>
```

3) expected attempt to access 
[https://github.com/DG12/add-longmem/blo](https://github.com/DG12/add-longmem/b)
to use my 404.html 
but the cute sandperson still shows up.

Any assistance would be appreciated .
Thank you

**Comment by airtower-luna** [comment]
Those instructions apply to a deployed Pages site (which would be under `dg12.github.io` for you), not to browsing non-existing files on the GitHub website. You can't set custom error pages for `github.com`. Allowing that would open some pretty big security holes, too. :sweat_smile:

**Comment by jfullstackdev** [comment]
if you can share the exact link of the repo, I can take a look, 

all I'm seeing at your profile are forks and I can't tell which is which

**Reply by DG12** [reply]
Of course.  [https://github.com/DG12/add-longmem ](https://github.com/DG12/add-longmem )

**Reply by jfullstackdev** [reply]
unfortunately, that's not public

![image](https://user-images.githubusercontent.com/47092464/190408148-56218c0b-d7fe-41d5-a551-76a44a66bcad.png)

**Reply by DG12** [reply]
Ooops. I will make it public. this is exactly the scenario I am trying to intercept with a better message.
404 is actually incorrect !

**Comment by DG12** [comment]
oops. it's public now.  This is exactly the scenario(private repository, user not logged) I am trying to intercept since 404 is really not correct.

**Comment by airtower-luna** [comment]
Aside from the issue I mentioned above (you can't set a custom error page for `github.com`, only your Pages site), the deployment workflow seems to [fail consistently](https://github.com/DG12/add-longmem/actions/runs/3057317749/jobs/4932347532).

The reason is that the [`textbook/git-checkout-submodule-action` action you use](https://github.com/DG12/add-longmem/blob/1f118e75a5f7926d909705a7010a50c6bf697d70/.github/workflows/gh-pages-push.yml#L25-L26) can't download a container image it needs. The action is archived and outdated anyway, you should [tell `actions/checkout` to get submodules](https://github.com/actions/checkout#usage) instead.

**Reply by DG12** [reply]
I didn't intend to set error page for GitHub.com.

I  intended to set error page  on to access my (private, [[temporarily made public]]) repository, 
especially discussions, when a user, not signed in to git hub attempts to access (any) page in my repository.

Thanks for that info about failing workflow

**Reply by airtower-luna** [reply]
The link you posted above as an example of where you expected to see your 404 page was a `github.com` one, that's why I pointed it out:

> expected attempt to access
> [https://github.com/DG12/add-longmem/blo](https://github.com/DG12/add-longmem/b)
> to use my 404.html
> but the cute sandperson still shows up.

If that's not what you meant I'd say fix the deployment, and see if the 404 page actually shows up.

As a side note, instead of pushing to `gh-pages` you might want to switch to the [new(-ish) custom deployment](https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/) (it's [out of beta](https://github.blog/changelog/2022-08-10-github-pages-builds-with-github-actions-ga/) by now).
